src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -1291,6 +1305,10 @@ private void Dispose(bool isDisposing)
             if (_isDisposed)
                 return;
 
+            var asyncTasksCounter = Interlocked.Read(ref _asyncTasksCounter);
+            if (asyncTasksCounter != 0)
+                throw new InvalidOperationException($"Disposing session with active async task is forbidden. Number of active async tasks: {asyncTasksCounter}");
```

Comment:
Can we improve the error message a bit and add:

```
Please make sure that all asynchronous session methods returning Task are awaited.
```

to be explicit about what needs to be checked.

Reply:
done